README.md

```diff
@@ -115,6 +115,39 @@ Server
 In many rails projects when you run the server with `rails s` Rails sets RAILS_ENV to "development".  If you do that with this app, you will connect to the development database on AWS (if you have permission to decrypt the value).  Instead, run `RAILS_ENV=local rails s` to start the server and `RAILS_ENV=local rails c` to run the console.
 
 
+Multithreading
```

Comment:
"Multithreading" is the wrong term to use here, I think. This is much more about asynchronous processing, that can happen at some point in the future, rather than "threading" which is about parallel processing (and where the results of the different threads usually are gathered up). 

"Asynchronous processing" might be a less loaded term.

Reply:
I used the word with purpose.  Asynchronous code is executed in rails in either one of two ways:
- asynchronous process runs in same thread
- asynchronous process runs in second thread, by some third-party magic that rails doesn't provide.

If all we wanted was asynchronous in the same thread, we would have used HttParty, like we do for patron email verification now.  But because we wanted to use multithreading, we worked on this epic.

Reply:
In what way is HTTParty asynchronous?

Reply:
If I understand this correctly, HttParty can be run multithreaded, on the more common RoR implementations, because it's an I/O operation (https://www.rubyguides.com/2015/07/ruby-threads/).  So, if you're using it to talk to Sierra, and the request takes a long time, you're probably going to be fine on one of our AWS Passenger servers.  However, HttParty is designed to be sure to be asynchronous, but not to be explicitly multithreaded.  So, you can have a list of asynchronous operations on the stack, and you check them every x milliseconds to see if it's time to do one.  (There is a good explanation I read a while back, I'm looking for the link.)  So, HttParty is always asynchronous, and usually multithreaded, but it's not guaranteed.  Some tasks are not considered I/O.  For those tasks, RoR doesn't do multithreading by default.  In addition, from what I understand, multithreading still happens within the same process on the CPU.  If you run such a task asynchronously/multithreaded on the same process as the main app, you'll risk slowing down the app.  

If we want to be sure of the multithreaded implementation that doesn't slow down the app and won't crash the app if it crashes, we need to make sure we're spinning our own worker with an ActiveJob.  That worker then spins a process, and there can be one or more threads on that process.

Originally, the functionality we wanted was to periodically crawl through rows of "incomplete" users, attempting to reconcile their barcodes.  I felt that this was a potentially slow and resource-intensive operation (albeit an I/O one), and it'd be safer to follow other Digital groups' example, and use ActiveJob.

Reply:
Okay, thanks for that.

Normally when we say a function or a method is asynchronous we mean that the function returns before the action is complete. This is the prevalent mode in node.js for example. But languages like Ruby & Python are synchronous, unless you explicitly say otherwise (using threads, for example).

To illustrate:

```
HTTParty.get 'https://nypl.org/'
puts 'next line'
```
when `next line` is printed the `get` method will have completed.

In JavaScript it tends to be asynchronous:
```
    xmlhttp = new XMLHttpRequest();
    xmlhttp.open("GET", 'https://nypl.org');
    xmlhttp.send(null);

    console.log('next line');
```
so `next line` will be printed _before_ the request has completed.

It's totally true that HTTP requests, like most IO operations, don't block at the OS level, so that they don't consume CPU cycles while they wait — but that is different from it being asynchronous at the application or code level. 

Threads are different — they are created within a single application or executable. So they're suitable for when you have two bits of code that can be executed in parallel, not for something where we want to do somewhat lengthy processing at a later time. It can be confusing, as people sometimes use 'thread' a little bit more generally, but "multithreading" in a technical context should always mean using actual threads at the application level. (As far as I can see HTTParty does not use threads at all.)

Here's a trivial example in Ruby:
```
t1 = Thread.new{func1()}
t2 = Thread.new{func2()}
t1.join
t2.join
```

(There are other ways of achieving asynchronous code too — event-driven execution, as node.js uses, is one.)

It seems to me that the crucial thing here is that we are registering ActiveJobs that a separate process looks for and processes. Those things _could_ use multithreading, but don't need to, and — as far as I can see — in this code are not.

Reply:
We're making a second worker.  That second worker creates a process.  The process then spins a thread.  Within that thread, DelayedJob is single-threaded -- all of the jobs are run in series, not parallel.  However, we do now have two threads -- the main app thread and the thread that DelayedJob's new process made.

This is a lot of back and forth on a small PR.  I'll change the verbiage.